src/shims/unix/unnamed_socket.rs

```diff
@@ -207,15 +225,16 @@ fn anonsocket_write<'tcx>(
 
 /// Read from AnonSocket and return the number of bytes read.
 fn anonsocket_read<'tcx>(
-    anonsocket: &AnonSocket,
-    peer_fd: Option<FileDescriptionRef>,
+    self_ref: WeakFileDescriptionRef,
+    peer_fd: WeakFileDescriptionRef,
```

Comment:
`peer_fd` doesn't need to be passed, you can always get it from the AnonSocket (preexisting issue)